File: languages/go/entity/availability/ipns_key.go
package availability

import (
	"github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"
)

// IpnsKey represents an object with a key uuid identifier.
type IpnsKey struct {
	KeyID string
}

// NewIpnsKey creates an IpnsKey instance with a key uuid identifier.
func NewIpnsKey(keyID string) IpnsKey {
	return IpnsKey{
		KeyID: keyID,
	}
}

func (s IpnsKey) ToProto() *proto.IpnsKey {
	return &proto.IpnsKey{
		KeyId: s.KeyID,
	}
}

func NewIpnsKeyFromProto(ipnsKey *proto.IpnsKey) *IpnsKey {
	if ipnsKey == nil {
		return nil
	} else {
		return &IpnsKey{
			KeyID: ipnsKey.KeyId,
		}
	}
}

File: languages/js/src/entity/availability/ipns_key.ts
import * as proto from "../../bridge/proto/bloock_availability_entities";

/**
 * Represents an object with a key uuid identifier.
 */
export class IpnsKey {
  keyID: string;

  /**
   * Creates an IpnsKey instance with a key uuid identifier.
   * @param keyID
   */
  constructor(
    keyID: string
  ) {
    this.keyID = keyID;
  }

  public toProto(): proto.IpnsKey {
    return proto.IpnsKey.fromPartial({
      keyId: this.keyID
    });
  }

  static fromProto(res: proto.IpnsKey): IpnsKey {
    return new IpnsKey(res.keyId);
  }
}
